### PR TITLE
DS-264 | @rebeccahongsf | add light form page component

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW

# Summary
- DS-264 | add light form page block to SB and scaffold component
- DS-265 | build out GG Nonce fetch

# Review By (Date)
- retroreview 
I might just roll this in and continue onto styling and megaprofile updates. Please feel free to request changes and share thoughts! I will make sure to implement any feedback in a future PR.

# Review Tasks

## Setup tasks and/or behavior to test

1. Navigate to https://deploy-preview-738--stanford-alumni.netlify.app/membership/payment-history
2. Verify that a payment form loads
3. Review code

# Associated Issues and/or People
- DS-264
- DS-265

### CODE REVIEW

File: src/components/page-types/lightFormPage/lightFormPage.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  const [kwoCreds, setKwoCreds] = useState('');
+
+  // Use the useEffect hook to fetch nonce when the component mounts
+  useEffect(() => {
+    // Function to fetch nonce from the API route
+    if (orgId && dssId) {
+      const fetchData = async () => {
+        try {
+          const response = await fetch(
+            `/api/membership/payment/${orgId}/${dssId}/2QZ7I82`
```
Comment (rebeccahongsf) [inline src/components/page-types/lightFormPage/lightFormPage.js:48]: Hardcoding the payment schedule id for now. I'll revisit configuring this with the payment schedule id from the userProfile in [DS-267](https://stanford.atlassian.net/browse/DS-267)

File: src/components/embed/giveGabForm.js [hunk 2]
```diff
@@ -27,6 +28,13 @@ const GiveGabForm = ({
     embedUrl.searchParams.set('urlData', urlData.replace(' ', ''));
   }
 
+  if (kwoCredentials) {
```
Comment (rebeccahongsf) [inline src/components/embed/giveGabForm.js:31]: I'm definitely wondering if there's a more general way we can append additional params to the script URL 🤔 

T/S and Membership use urlData for trips and promo/appeal codes.

File: src/components/embed/dynaScript.js [hunk 3]
```diff
@@ -38,6 +38,9 @@ const DynaScript = ({ errorBlok, src, id, ...props }) => {
       if (mounted) {
         setScriptLoaded(true);
 
+        // @TODO: Temp workaround for 'widgetRenderEnd' event listener
+        if (src.href.includes('kwoCredentials')) showForm();
+
         // Once GiveGab form has completed rendering, display form
         script.addEventListener('widgetRenderEnd', showForm);
```
Comment (rebeccahongsf) [inline src/components/embed/dynaScript.js:45]: For some reason the widgetRenderEnd is not firing so I added this workaround for now. Ashleigh was able to test and confirm that the following widget were available in [Basecamp](https://3.basecamp.com/3590034/buckets/30128905/messages/6661562881#__recording_6669917089):

- `widgetDefinitionReceived`
- `widgetRenderStart`
- `widgetRenderEnd`

 I'll circle back to debug later on.

File: src/api/membership/payment/[...history].js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+const paymentHandler = async (req, res) => {
+  // Extract orgId and dssId from the URL parameters
+  const { history } = req.params;
+  const [orgId, dssId, scheduleId] = history.split('/');
+  const ggNonce = new GiveGabNonce();
+
+  try {
+    // Use orgId and dssId to fetch the nonce token
+    const nonceToken = await ggNonce.getToken(orgId, dssId, scheduleId);
+    res.json({ ...nonceToken });
```
Comment (rebeccahongsf) [inline src/api/membership/payment/[...history].js:12]: The nonce is fetch every time the light form page is loaded. Should we consider caching it until it expires? 🤔

Comment (sherakama) [reply src/api/membership/payment/[...history].js:12]: You could add it to the user's cookies but what is this nonce being used for? Why do you need it?

Comment (rebeccahongsf) [reply src/api/membership/payment/[...history].js:12]: The nonce is needed for the GiveGab self-service widget embed (additional details in this [basecamp thread](https://3.basecamp.com/3590034/buckets/30128905/documents/6496437183)).

**Pull request merged**